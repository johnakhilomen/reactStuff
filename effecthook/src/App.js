import {useEffect, useState} from "react";

import './App.css';
import ProfileEdit from "./ProfileEdit";

function App() {
  //Load via useEffect only once
  const [user, setUser] = useState({})
  const [time, setTime] = useState("")
  
  useEffect(() => {
    async function loadUsers() {
      const response = await fetch(
        "https://jsonplaceholder.typicode.com/users/1"
      );
      const userFromAPI = await response.json();
      setUser(userFromAPI);
    }
    loadUsers();
  }, []); // Passing [] so that it only runs the effect once


  useEffect(() => {
    setTime(`The time is now ${Date.now()}`);
  });


  if (user.id) {
    // `user.id` is truthy after the API call returns
    return (
      <div>
        <form name="profileEdit">
        <div>
          <label htmlFor="username">User Name:</label>
          <input
            id="username"
            name="username"
            type="text"
            defaultValue={user.username}
          />
        </div>
        <div>
          <label htmlFor="email">Email:</label>
          <input id="email" name="email" type="email" defaultValue={user.email} />
        </div>
      </form>
      <br></br>
      <p>edit profile...</p>
      <ProfileEdit></ProfileEdit>
      </div>
    );
  }
  return "Loading...";
}

export default App;
