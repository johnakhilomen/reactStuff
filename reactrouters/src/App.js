import React from "react";
import { BrowserRouter as Router, Route } from "react-router-dom";
import UserProfile from "./UserProfile";

function Home() {
  return <p>Home</p>;
}

function About() {
  return <p>About</p>;
}

function App() {
  return (
    <Router>
      <div className="App">
        <Route exact={true} path="/">
          <Home />
        </Route>
        <Route path="/about">
            <About></About>
        </Route>
        <Route path="/userprofile/:id">
          <UserProfile></UserProfile>
        </Route>
      </div>
    </Router>
  );
}

export default App;