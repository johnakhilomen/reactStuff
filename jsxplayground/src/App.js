import logo from './logo.svg';
import './App.css';
import Header from './Header';
import Table from './Table';

function App() {
  return (
    <div className="App">
      <Header></Header>
      <Table></Table>
    </div>
  );
}

export default App;
