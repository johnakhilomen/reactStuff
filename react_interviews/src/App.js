import logo from "./logo.svg";
import "./App.css";
import { useCallback, useMemo, useState } from "react";

function App() {
  /* 
Virtual dom : plain JavaScript Object that represents the tag to be rendered onto the screen.
*/
  console.log(<div />);
  console.log(<div p={{ name: "prop" }} />);
  // or
  // console.log(React.createElement('div', null))

  /* 
  This technique HOC (Higher Order Component) is used to pass elements as props children
  */

  /* 
The ref provides access to the DOM Element represented by a React Element.
  */

  /*
using React.PropTypes, react can restrict props to certain types 
*/

  /*
Pure components are components with no state, or lifecycle methods.
*/

  /* Prop drilling is the coupling of nested components in a messy manner which 
allows the exposure of data to other components which is disadvantagous 
when we don't want certain components to have access to the data
*/

  return <div className="App"></div>;
}

export default App;
