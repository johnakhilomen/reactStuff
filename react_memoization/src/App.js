import logo from "./logo.svg";
import "./App.css";
import ParentClassWithMemoization from "./ParentClassNoMemoization";
import ParentClassNoMemoization from "./ParentClassNoMemoization";

function App() {
  return (
    <div className="App">
      <ParentClassNoMemoization />
      {/* <ParentClassWithMemoization /> */}
    </div>
  );
}

export default App;
