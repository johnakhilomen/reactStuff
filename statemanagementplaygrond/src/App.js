import logo from './logo.svg';
import {useState} from "react";
import data from "./data";

import './App.css';
import Form from './Form';
import List from './List';

function App() {
  const [dataLocal, setDataLocal] = useState(data);
  return (
    <div className="App">
      <Form></Form>
    </div>
  );
}

export default App;
