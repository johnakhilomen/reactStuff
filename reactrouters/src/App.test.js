import React from "react";
import App from "./App";
import { shallow } from "enzyme";

//App
it("renders APP component without crashing", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
