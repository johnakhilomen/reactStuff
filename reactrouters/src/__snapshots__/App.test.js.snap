// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders APP component without crashing 1`] = `ShallowWrapper {}`;
